### DISCUSSION TITLE
Temperature and humidity sensor

### DISCUSSION THREAD

**Discussion by piokacz** [milo526/homebridge-tuya-web#154] [Q&A] [answered]
Hi,
I'm using the zigbee sensor, but im not able to push it to homebridge - other devices (also zigbee) are listed in homebridge. Below my configuration:

                    "id": "b******************j",
                    "device_type": "temperature_sensor"

Are those devices supported by 0.7.1 version of this plugin?

**Comment by milo526** [comment, marked as answer]
Please post the result of [the steps under "How to check whether the api this library uses can control your device".](https://github.com/milo526/homebridge-tuya-web#how-to-check-whether-the-api-this-library-uses-can-control-your-device)

Without this it is hard for me to help you with regards to this specific issue.

**Reply by piokacz** [reply]
Hi, this is the answer from repl.it:

Getting credentials
Got credentials
{   'access_token': 'EUhe___________________m94j',
    'expires_in': 864000,
    'refresh_token': 'EUh___________________kKy',
    'token_type': 'bearer'}
Getting devices
Got devices
{   'header': {'code': 'SUCCESS', 'payloadVersion': 1},
    'payload': {   'devices': [   {   'data': {'online': True, 'state': True},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/1541647698myzx2yx1yk_0.jpg',
                                      'id': '4780___________________515be',
                                      'name': 'Smart Socket'},
                                  {   'data': {'online': True, 'state': False},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/1541647698myzx2yx1yk_0.jpg',
                                      'id': '478___________________4f4aa',
                                      'name': 'Biurko'},
                                  {   'data': {'online': True, 'state': True},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/program_category_icon/kg.png',
                                      'id': '6666___________________ef9',
                                      'name': 'Super Mini Switch Module'},
                                  {   'data': {'online': True, 'state': True},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/1541647698myzx2yx1yk_0.jpg',
                                      'id': '478___________________ecc',
                                      'name': 'Gniazdo 2'},
                                  {   'data': {'online': True, 'state': True},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/1541647698myzx2yx1yk_0.jpg',
                                      'id': '478___________________46',
                                      'name': 'Akwarium'},
                                  {   'data': {'online': False, 'state': True},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/solution/76001/b8bb0466b861046d_cover.png',
                                      'id': 'bf4f___________________wp',
                                      'name': 'Włącznik łazienka góra '},
                                  {   'data': {   'current_temperature': 215,
                                                  'max_temper': 350,
                                                  'min_temper': 50,
                                                  'online': True,
                                                  'state': 'true',
                                                  'temperature': 210},
                                      'dev_type': 'climate',
                                      'ha_type': 'climate',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/ay1539244260287RY0l6/1559184771a8dbbfcbcbd.png',
                                      'id': 'bfe___________________i1',
                                      'name': 'Termostat salon'},
                                  {   'data': {'online': True, 'state': False},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/program_category_icon/kg.png',
                                      'id': '66___________________9_1',
                                      'name': 'kinkiet'},
                                  {   'data': {'online': True, 'state': False},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/program_category_icon/kg.png',
                                      'id': '667___________________99_2',
                                      'name': 'wentylator'},
                                  {   'data': {},
                                      'dev_type': 'scene',
                                      'ha_type': 'scene',
                                      'id': 'ev___________________1V',
                                      'name': 'If Temperature and humidity '
                                              'sensor Humidity : Mniej niż 67% '
                                              'RH,\\"Łazienka oświetlenie i w'},
                                  {   'data': {},
                                      'dev_type': 'scene',
                                      'ha_type': 'scene',
                                      'id': '8trb___________________pS',
                                      'name': 'If Temperature and humidity '
                                              'sensor Humidity : Wyższy niż '
                                              '70% RH,\\"Łazienka oświetlenie '
                                              'i '},
                                  {   'data': {},
                                      'dev_type': 'scene',
                                      'ha_type': 'scene',
                                      'id': 'lBu___________________KPb',
                                      'name': 'wentylator'},
                                  {   'data': {},
                                      'dev_type': 'scene',
                                      'ha_type': 'scene',
                                      'id': '0s___________________p6',
                                      'name': 'kinkiet'},
                                  {   'data': {},
                                      'dev_type': 'scene',
                                      'ha_type': 'scene',
                                      'id': 'W___________________cm',
                                      'name': 'lazienka s'},
                                  {   'data': {},
                                      'dev_type': 'scene',
                                      'ha_type': 'scene',
                                      'id': 'uZ___________________31',
                                      'name': '\\"TV\\" power'},
                                  {   'data': {},
                                      'dev_type': 'scene',
                                      'ha_type': 'scene',
                                      'id': 'sE___________________gy',
                                      'name': 'Temperatura 16'}],
                   'scenes': []}}
!!! NOW REMOVE THIS FILE, SO YOUR CREDENTIALS (username, password) WON'T LEAK !!!

I don't se my sensor.

**Reply by milo526** [reply]
So sadly your sensor is not supported by the API that this plugin is using and will not be exposed to Homebridge/HomeKit.

This is a limitation from Tuya and not something I can change.

**Reply by piokacz** [reply]
Thanks for your answer! We can delete this discussion :-)
